src/mercury_engine_data_structures/type_lib.py

```diff
@@ -202,97 +203,100 @@ def decode_type(name: str, data: dict) -> BaseType:
     return kind.type_class.from_json(name, data)
 
 
-@functools.lru_cache
-def all_types() -> Dict[str, BaseType]:
-    return {
-        name: decode_type(name, data)
-        for name, data in dread_data.get_raw_types().items()
-    }
+class TypeLib:
+    def __init__(self, game: Game):
```

Comment:
I'd make this take the dict that `get_raw_types()` returns directly as an argument. This would make this file not need any of the imports, which would make create_class_definition not need code replacement for it to work.